internal/config/types.go

```diff
@@ -25,103 +25,116 @@ func parseServerType(str string) (ServerType, bool) {
-	File               *File            `yaml:"-" mapstructure:"file"`
-	Common             *CommonSettings  `yaml:"-"`
-	Watchers           *Watchers        `yaml:"-"`
-}
+const (
+	Unsupported OTelReceiver = "unknown"
+	OTLP        OTelReceiver = "otlp"
+	HostMetrics OTelReceiver = "hostmetrics"
+)
 
-type Log struct {
-	Level string `yaml:"-" mapstructure:"level"`
-	Path  string `yaml:"-" mapstructure:"path"`
-}
+type (
+	Config struct {
+		UUID               string           `yaml:"-"`
+		Version            string           `yaml:"-"`
+		Path               string           `yaml:"-"`
+		Log                *Log             `yaml:"-" mapstructure:"log"`
+		DataPlaneConfig    *DataPlaneConfig `yaml:"-" mapstructure:"data_plane_config"`
+		Client             *Client          `yaml:"-" mapstructure:"client"`
+		ConfigDir          string           `yaml:"-" mapstructure:"config-dirs"`
+		AllowedDirectories []string         `yaml:"-"`
+		Metrics            *Metrics         `yaml:"-" mapstructure:"metrics"`
+		Command            *Command         `yaml:"-" mapstructure:"command"`
+		File               *File            `yaml:"-" mapstructure:"file"`
+		Common             *CommonSettings  `yaml:"-"`
+		Watchers           *Watchers        `yaml:"-"`
+	}
 
-type DataPlaneConfig struct {
-	Nginx *NginxDataPlaneConfig `yaml:"-" mapstructure:"nginx"`
-}
+	Log struct {
+		Level string `yaml:"-" mapstructure:"level"`
+		Path  string `yaml:"-" mapstructure:"path"`
+	}
 
-type NginxDataPlaneConfig struct {
-	ReloadMonitoringPeriod time.Duration `yaml:"-" mapstructure:"reload_monitoring_period"`
-	TreatWarningsAsError   bool          `yaml:"-" mapstructure:"treat_warnings_as_error"`
-}
+	DataPlaneConfig struct {
+		Nginx *NginxDataPlaneConfig `yaml:"-" mapstructure:"nginx"`
+	}
 
-type Client struct {
-	Timeout             time.Duration `yaml:"-" mapstructure:"timeout"`
-	Time                time.Duration `yaml:"-" mapstructure:"time"`
-	PermitWithoutStream bool          `yaml:"-" mapstructure:"permit_without_stream"`
-}
+	NginxDataPlaneConfig struct {
+		ReloadMonitoringPeriod time.Duration `yaml:"-" mapstructure:"reload_monitoring_period"`
+		TreatWarningsAsError   bool          `yaml:"-" mapstructure:"treat_warnings_as_error"`
+	}
 
-type Metrics struct {
-	// temporary setting to enable the collector
-	Collector bool `yaml:"-" mapstructure:"collector"`
-}
+	Client struct {
+		Timeout             time.Duration `yaml:"-" mapstructure:"timeout"`
+		Time                time.Duration `yaml:"-" mapstructure:"time"`
+		PermitWithoutStream bool          `yaml:"-" mapstructure:"permit_without_stream"`
+	}
 
-type GRPC struct {
-	Target         string        `yaml:"-" mapstructure:"target"`
-	ConnTimeout    time.Duration `yaml:"-" mapstructure:"connection_timeout"`
-	MinConnTimeout time.Duration `yaml:"-" mapstructure:"minimum_connection_timeout"`
-	BackoffDelay   time.Duration `yaml:"-" mapstructure:"backoff_delay"`
-}
+	Metrics struct {
```

Comment:
I was thinking metrics would follow a similar structure to Commad?
with an exporter list sub entry in this case. Then Server, Auth  and TLS under it? 

Perhaps something like:
```
metrics:
    collector: true,
    exporters:
        - otlp (default)
          server:
            ...
          auth: 
           ...
          tls: 
          ....
``` 
or alternatively?

Reply:
We can do that, sure. Let me see what I can do.

Reply:
Okay the structure of `Collector` now follows the suggested format 👍 took a minute to get the `viper.Unmarshal` magic to work for unmarshalling full `mapstructure` structs.